porch/controllers/workloadidentitybinding/pkg/controllers/workloadidentitybinding/controller.go

```diff
@@ -211,9 +230,55 @@ func (r *WorkloadIdentityBindingReconciler) BuildObjectsToApply(ctx context.Cont
 		})
 		obj.SetOwnerReferences(ownerRefs)
 	}
+
 	return objects, nil
 }
 
+func (r *WorkloadIdentityBindingReconciler) addWiAnnotation(ctx context.Context, projectID string, subject *api.WorkloadIdentityBinding) error {
+	sa, err := r.getServiceAccount(ctx, subject)
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	gsaEmail := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", subject.Spec.ResourceRef.Name, projectID)
```

Comment:
I think we should to look up the referenced IAMServiceAccount and get the email from the status.  I don't think we can assume the project-id or the name.

Reply:
Updated the PR to look up the IAMServiceAccount. As part of this we also change the field names in the CRD, so this is a very breaking change.